go/src/net/Connection.go

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package nebulaNet
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/facebook/fbthrift/thrift/lib/go/thrift"
+	graph "github.com/vesoft-inc/nebula-clients/go/nebula/graph"
+	conf "github.com/vesoft-inc/nebula-clients/go/src/conf"
+	data "github.com/vesoft-inc/nebula-clients/go/src/data"
+)
+
+type Connection struct {
+	SeverAddress data.HostAddress
+	graph        *graph.GraphServiceClient
+	isValid      bool
+}
+
+func NewConnection(severAddress data.HostAddress) *Connection {
+	newObj := Connection{}
+	newObj.SeverAddress = severAddress
+	newObj.graph = nil
+	return &newObj
+}
+
+func (cn *Connection) Open(hostAddress data.HostAddress, conf conf.PoolConfig) (err error) {
+	ip := hostAddress.GetHost()
+	port := hostAddress.GetPort()
+	newAdd := ip + ":" + strconv.Itoa(port)
+	timeoutOption := thrift.SocketTimeout(conf.TimeOut)
+	addressOption := thrift.SocketAddr(newAdd)
+	sock, err := thrift.NewSocket(timeoutOption, addressOption)
+	if err != nil {
+		log.Printf("Failed to create a net.Conn-backed Transport,: %s", err.Error())
+		return err
+	}
+
+	transport := thrift.NewBufferedTransport(sock, 128<<10)
+
+	pf := thrift.NewBinaryProtocolFactoryDefault()
+	cn.graph = graph.NewGraphServiceClientFactory(transport, pf)
+
+	if err := cn.graph.Transport.Open(); err != nil {
+		log.Printf("Failed to open transport, error: %s", err.Error())
+		return err
+	}
+	if cn.graph.Transport.IsOpen() == false {
+		log.Printf("Transport is off: \n")
+	}
+	return nil
+}
+
+// Open transport and authenticate
+func (cn *Connection) Authenticate(username, password string) (*graph.AuthResponse, error) {
+	resp, err := cn.graph.Authenticate([]byte(username), []byte(password))
+	if err != nil {
+		log.Printf("Authentication fails, %s", err.Error())
+		if e := cn.graph.Close(); e != nil {
+			log.Printf("Fail to close transport, error: %s", e.Error())
+		}
+		return nil, err
+	}
+
+	if resp.GetErrorCode() != graph.ErrorCode_SUCCEEDED {
```

Comment:
if you return the resp, here not need to check the error code, let the outside to check it